lib/Rx/Observable/TimerObservable.php

```diff
@@ -25,11 +26,10 @@ public function __construct($dueTime, SchedulerInterface $scheduler = null)
         $this->scheduler = $scheduler;
     }
 
-    public function subscribe(ObserverInterface $observer, SchedulerInterface $scheduler = null)
+    public function subscribe(ObserverInterface $observer)
     {
-        if ($this->scheduler !== null) {
-            $scheduler = $this->scheduler;
-        }
+        $scheduler = $this->scheduler ?: Scheduler::getDefault();
```

Comment:
I think it is important to have a ```$scheduler``` parameter on the constructor of time-based operators. If an unsuspecting user subscribes to an observable that uses timing, they may not know to send the appropriate scheduler.